apps/alarm/request.js

```diff
@@ -14,51 +14,67 @@ function _createMd5 (extraData) {
     .toUpperCase()
 }
 
-function request (params) {
+function request (params, requestCallback) {
   params.activity.get()
     .then(extraData => {
-      id++
-      var data = {
-        deviceId: extraData.deviceId,
-        deviceType: extraData.deviceTypeId,
-        intent: params.intent,
-        masterId: extraData.masterId,
-        nonce: extraData.deviceId + (new Date()).getTime() + id,
-        requestTimestamp: (new Date()).getTime(),
-        sessionId: (new Date()).getTime(),
-        sign: _createMd5(extraData),
-        signMethod: 'MD5',
-        businessParams: params.businessParams || {}
-      }
-      data = Buffer.from(JSON.stringify(data))
-      var req = https.request({
-        method: 'POST',
-        host: DEFAULT_HOST,
-        path: DEFAULT_URI || params.url,
-        headers: {
-          'Content-Type': 'application/json',
-          'Content-Length': data.length
+      try {
+        id++
+        var data = {
+          deviceId: extraData.deviceId,
+          deviceType: extraData.deviceTypeId,
+          intent: params.intent,
+          masterId: extraData.masterId,
+          nonce: extraData.deviceId + (new Date()).getTime() + id,
+          requestTimestamp: (new Date()).getTime(),
+          sessionId: (new Date()).getTime(),
+          sign: _createMd5(extraData),
+          signMethod: 'MD5',
+          businessParams: params.businessParams || {}
         }
-      }, (res) => {
-        var list = []
-        res.on('data', (chunk) => list.push(chunk))
-          .on('end', () => {
-            var result = Buffer.concat(list).toString()
-            if (res.statusCode !== 200) {
-              logger.error(`Error: failed get data with ${result}`)
-            } else {
-              if (typeof params.callback === 'function') {
-                params.callback(result)
+        data = Buffer.from(JSON.stringify(data))
+        var req = https.request({
+          method: 'POST',
+          host: DEFAULT_HOST,
+          path: DEFAULT_URI || params.url,
+          headers: {
+            'Content-Type': 'application/json',
+            'Content-Length': data.length
+          }
+        }, (res) => {
+          var list = []
+          res.on('data', (chunk) => list.push(chunk))
+            .on('end', () => {
+              var result = Buffer.concat(list).toString()
+              if (res.statusCode !== 200) {
+                logger.error(`Error: failed get data with ${result}`)
+                if (requestCallback) {
+                  requestCallback(false)
+                }
+              } else {
+                if (typeof params.callback === 'function') {
```

Comment:
What's the difference between `params.callback` and `requestCallback`?

Reply:
params.callback is callback for request result data ,requestCallback just for request success or not, don't care the request result data, requestAlarms function contain sync_alarm/upload_alarms request,just upload_alarms need use requestCallback, and sync_alarm/upload_alarms need params.callback ,so add requestCallback as a param for requestAlarms